src/App.tsx

```diff
@@ -1,26 +1,236 @@
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import * as TodosService from './api/todos';
+import classNames from 'classnames';
+import { TodoList } from './TodoList';
+import { TodoStatusFilter } from './types/TodoStatusFilter';
+import { TodoItem } from './TodoItem';
+import { NewTodo } from './NewTodo';
+import { Filter } from './Filter';
 
-const USER_ID = 0;
+function getFilteredTodos(todos: Todo[], status: TodoStatusFilter) {
+  let preparedTodos = todos;
+
+  if (status === TodoStatusFilter.Completed) {
+    preparedTodos = preparedTodos.filter(todo => todo.completed);
+  } else if (status === TodoStatusFilter.Active) {
+    preparedTodos = preparedTodos.filter(todo => !todo.completed);
+  }
+
+  return preparedTodos;
+}
 
 export const App: React.FC = () => {
+  const field = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState('');
+  const [statusFilter, setStatusFilter] = useState<TodoStatusFilter>(
+    TodoStatusFilter.All,
+  );
+  const [todoTitle, setTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [blockInput, setBlockInput] = useState(false);
+  const [deletingTodos, setDeletingTodos] = useState<number[]>([]);
+  const [displayError, setDisplayError] = useState(false);
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+
+  const hasCompletedTodos = todos.some(todo => todo.completed);
+
+  useEffect(() => {
+    setLoading(true);
+
+    TodosService.getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+        setDisplayError(true);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (!error) {
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setDisplayError(false);
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [error]);
+
+  const handleAddTodo = () => {
+    const normalizedTitle = todoTitle.trim();
+
+    if (!normalizedTitle) {
+      setError('Title should not be empty');
+      setDisplayError(true);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title: normalizedTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+    setBlockInput(true);
+
+    TodosService.addTodo({
+      title: normalizedTitle,
+      userId: USER_ID,
+      completed: false,
+    })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTodoTitle('');
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+        setDisplayError(true);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setBlockInput(false);
+      });
+  };
+
+  function deleteTodo(id: number) {
```

Comment:
```suggestion
  function handleDeleteTodo(id: number) {
```
Be consistent with the naming